src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,98 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private Object[] storage = new Object[INITIAL_CAPACITY];
+    private int currentTailIndex = 0;
```

Comment:
```suggestion
    private int currentTailIndex;
```

0 is default value for int